circleguard/widgets.py

```diff
@@ -694,20 +652,46 @@ def set_dir(self):
         if update_path != "":
             self.update_dir(update_path)
 
-
     def update_dir(self, path):
         self.path = path if path != "" else self.path
+        self.changed = True if self.path != self.default_path else False
         if self.display_path:
             label = path if len(self.path) < 64 else ntpath.basename(path)
             self.path_label.setText(label)
         self.path_signal.emit(self.path)
 
-
     def switch_enabled(self, state):
         self.label.setStyleSheet("color:grey" if not state else "")
         self.path_label.setStyleSheet("color:grey" if not state else "")
         self.file_chooser_button.setEnabled(state)
 
+    def show_required(self):
+        self.combined.setStyleSheet(get_setting("required_style"))
+        self.highlighted = True
+
+    def mousePressEvent(self, event):
+        super().mousePressEvent(event)
+        self.reset_highlight()
+
+    # separate function so we can call this method outside of mousePressEvent
+    def reset_highlight(self):
+        if self.highlighted:
+            self.combined.setStyleSheet(self.old_stylesheet)
+            self.highlighted = False
+
+class BorderWidget(QFrame):
+    """
+    For debugging widgets that don't seem to be appearing. Adds a white border
+    around the QFrame.
+
+    To use on a class, replace the inheritence of QFrame with BorderWidget.
+    Possibly works with multiple inheritence (using this as a mixin), but I'm
+    not sure about that.
+    """
+    def __init__(self):
+        super().__init__()
+        self.setFrameStyle(QFrame.Box)
+
```

Comment:
remove debug class?

Reply:
could be useful in the future? I've already used, then removed, a class very similar to this twice when debugging. I'll probably just end up forgetting about it and re-looking up how to set a box frame. And it's really one doing one line of work, so...will remove